### PR TITLE
Upgrading CI scripts.

### PR DESCRIPTION
**Pull request by armfazh**
Upgrading CI scripts.

### CODE REVIEW

File: package.json
Comment (thibmeu) [inline package.json:1]: consider bumping to node.js 20

```
"engines": {
        "node": ">=16"
    },
```

**Comment by thibmeu** [comment]
change is good overall. minor nits.

File: package.json [hunk 1]
```diff
@@ -50,6 +52,7 @@
         "test": "tsc -b test && node --experimental-vm-modules ./node_modules/jest/bin/jest.js --ci --no-cache",
         "lint": "eslint .",
         "format": "prettier './(src|test)/*.ts' --write",
+        "check-format": "prettier './(src|test)/*.ts' --check",
```
[the remark this answers is not in the document]
Comment (armfazh) [reply package.json]: correct, I just added to it.

File: package.json
Comment (armfazh) [reply package.json:1]: done

**Pull request merged**